apps/vs-code-extension/src/standard-library-file-system-provider.ts

```diff
@@ -0,0 +1,80 @@
+// SPDX-FileCopyrightText: 2023 Friedrich-Alexander-Universitat Erlangen-Nurnberg
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+
+import { STANDARD_LIBRARY_SOURCECODE } from '@jvalue/jayvee-language-server';
+import {
+  EventEmitter,
+  ExtensionContext,
+  FileChangeEvent,
+  FileStat,
+  FileSystemError,
+  FileSystemProvider,
+  FileType,
+  Uri,
+  workspace,
+} from 'vscode';
+
+export class StandardLibraryFileSystemProvider implements FileSystemProvider {
+  // The following class members only serve to satisfy the interface:
+  private readonly didChangeFile = new EventEmitter<FileChangeEvent[]>();
+  onDidChangeFile = this.didChangeFile.event;
+
+  static register(context: ExtensionContext) {
+    context.subscriptions.push(
+      workspace.registerFileSystemProvider(
+        'builtin',
+        new StandardLibraryFileSystemProvider(),
+        {
+          isReadonly: true,
+          isCaseSensitive: false,
+        },
+      ),
+    );
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  stat(uri: Uri): FileStat {
+    const date = Date.now();
+    return {
+      ctime: date,
+      mtime: date,
+      size: STANDARD_LIBRARY_SOURCECODE.length,
```

Comment:
Would have expected this is the byte size of the document? Then we'd have to do some calculation like each character times bytes per character.